### ISSUE TITLE
Coop workers should be able to add tasks

### ISSUE THREAD

**Issue by bhaugen** [FreedomCoop/valuenetwork#65]
Rename Todos (the simple OCP tasks) to be Task. Coop workers should be able to add such tasks to their groups, and assign themselves or other members of that group.

**Comment by enricduran** [comment]
when they create a task, coop workers only should see the context and types related the groups they are

**Comment by fosterlynn** [comment]
Clarification:  There are 2 types of tasks, 1) the to-do's and 2) the various work needed to execute a process.  You need a process if you are going to create something that will be traded (sold) or used in another process.  So most tasks are generally related to a process.  The to-do's were originally designed for quick small random tasks not related to the main production activities, but things you need to remember to do, or things you want to quickly request someone else to do.

Of course we can change things.  But the implication of using to-do's a lot is that the work done is not part of a value flow, it is within a project but otherwise disconnected.  So in a value equation, it will have to be picked up by a date range and project, rather than being found related to a deliverable or final produced output or customer order.

Questions: What does it mean "types"? Is this types of work, also called skills?

**Comment by bum2** [comment]
Thanks for the clarification.

> Questions: What does it mean "types"? Is this types of work, also called skills?
I think so Lynn. The point for 'coop workers' (roles "FC Members", "Candidates" and "Project Participants") is to see only the things related to them. If the project defines a set of types (of task, of resource, of exchange, etc) for their needs, then the workers of the project only choose among those types. 
Also the 'context' of a task can be fixed to the project or organization creating or needing the task (not choosed) in pre-launch. After we'll study use cases where context can vary, and try to define the possibilities from the project or agent type level.
See the roles in issue #80

**Comment by bhaugen** [comment]
@bum2 as of now, types of work are not defined by project. They are global for all of OCP.

We could change that and define them by project, but then most likely many projects would define the same types of work. And then each project needs to start by defining types of work, which they may not know how to think about. 

I think it would be good to have some global types of work and then allow projects to define their own once they need some that might be restricted to that project.

**Comment by bum2** [comment]
Yes @bhaugen, type of work are global. I mean that the types of work the project has choosed in their tasks are together the subset of types the participant of that project can choose from.
Post -launch is the 'Other' field to open and moderate the inclusion of new types (types of work in this case)

**Comment by bhaugen** [comment]
We can also filter types of work by use case, which @fosterlynn or I can show people how to do when they need it.

**Comment by bhaugen** [comment]
See comment on #71 and try the work dashboard after @XaviP gets my last commit deployed.

Questions: should workers with a candidate but not full membership be allowed to post or take tasks? How about project participants?

**Comment by bum2** [comment]
Yes, candidates and participants can post simple tasks and take simple, planned or un-planned tasks (and get paid for that).

**Comment by bhaugen** [comment]
Excellent! That's how I made it.
